wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js

```diff
@@ -32,59 +82,86 @@ function setupModal(modal, jsonData) {
      * Search the list
      */
     var searchForm = modal.body.find('form.search-bar');
-    var searchField = searchForm.find('#id_q');
     var searchUrl = searchForm.attr('action');
     var searchResults = modal.body.find('#search-results');
+    // Builds a dictionary of the values of fields in searchForm
+    function buildSearchDict() {
+        var rtn = {};
+        searchForm.find("input, select, textarea").each(function() {
+            rtn[$(this).attr("name")] = $(this).val();
```

Comment:
I don't think this covers boolean form controls like checkboxes and radios, value will be the value not which on is checked.

You could possibly use the `FormData` class to easily build up a dict

https://developer.mozilla.org/en-US/docs/Web/API/FormData

Reply:
Thanks for the pickup. Had a go with FormData, but the API is weird. Added a case for radios/checkboxes using jQuery.